src/expedited.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-2017 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_EXPEDITED_H
+#define BITCOIN_EXPEDITED_H
+
+#include "net.h"
+#include "thinblock.h"
+
+#include <univalue.h>
+#include <vector>
+
+
+enum
+{
+    EXPEDITED_STOP = 1,
+    EXPEDITED_BLOCKS = 2,
+    EXPEDITED_TXNS = 4,
+};
+
+enum
+{
+    EXPEDITED_MSG_HDR = 1,
+    EXPEDITED_MSG_XTHIN = 2,
+};
+
+extern CCriticalSection cs_xpedited;
```

Comment:
Can you add a comment as to what precisely this critical section is intended to be protecting?